controllers/managedclusterForCGU_controller.go

```diff
@@ -86,12 +95,16 @@ func (r *ManagedClusterForCguReconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{}, err
 	}
 
-	// Stop creating UOCR if ztp of this cluster is done already
+	// Stop creating UOCR or AZTP if ztp of this cluster is done already
 	if _, found := managedCluster.Labels[ztpDoneLabel]; found {
 		r.Log.Info("ZTP for the cluster has completed. "+ztpDoneLabel+" label found.", "Name", managedCluster.Name)
 		return doNotRequeue(), nil
 	}
 
+	if err := r.extractAztpPolicies(ctx, managedCluster); err != nil {
```

Comment:
Does it make more sense to do this inside `handleNotReadyCluster`?

Reply:
Also, not sure why the available condition check was written that way. It could have been 
```
if availableCondition == nil || availableCondition.Status == metav1.ConditionFalse {
    return r.handleNotReadyCluster(managedCluster), nil
} else {
...
}
```

Reply:
Initially, it had different logs for condition=nil and status=false cases. As they were removed,  the checks can be simply merged.

Reply:
If it goes into handleNotReadyCluster, it could be either creation case or cluster becomes NotReady again right after it gets reqeued(ubnormal case). 
I am not familiar with the background of this story. Not sure the reason of redo policy extraction? Is it because of the conern where the child policies have not present yet when it initially gets requeued?

Reply:
Yep, I think it makes sense to move it to handleNotReadyCluster. 
For the abnormal case, there is reason, but IMHO also no harm in redoing the policy extraction

Reply:
fixed